src/Collections/Maps/Maps.dfy

```diff
@@ -21,15 +21,19 @@ module Maps {
   }
 
   function method {:opaque} ToImap<X, Y>(m: map<X, Y>): (m': imap<X, Y>)
+    /* Dafny selected triggers: {m[x]}, {m'[x]}, {x in m'}, {x in m} */
     ensures forall x {:trigger m'[x]} :: x in m ==> x in m' && m'[x] == m[x]
+    /* Dafny selected triggers: {x in m}, {x in m'} */
     ensures forall x {:trigger x in m'} :: x in m' ==> x in m
   {
     imap x | x in m :: m[x]
   }
 
   /* Remove all key-value pairs corresponding to the set of keys provided. */
   function method {:opaque} RemoveKeys<X, Y>(m: map<X, Y>, xs: set<X>): (m': map<X, Y>)
+    /* Dafny selected triggers: {m[x]}, {m'[x]}, {x in m'}, {x in xs}, {x in m} */
     ensures forall x {:trigger m'[x]} :: x in m && x !in xs ==> x in m' && m'[x] == m[x]
+    /* Dafny selected triggers: {x in xs}, {x in m}, {x in m'} */
     ensures forall x {:trigger x in m'} :: x in m' ==> x in m && x !in xs
```

Comment:
Since the ensures is an implication, we might only want the `x in m'`? My limited understanding is that we are ensuring the properties of `m'`, rather than the original `m`. So might not be necessary to invoke this fact every time we see the original map.